app/src/main/java/bg/zahov/app/ui/workout/WorkoutScreen.kt

```diff
@@ -65,22 +65,22 @@ import androidx.compose.ui.text.input.VisualTransformation
 import androidx.compose.ui.text.style.TextOverflow
 import androidx.compose.ui.unit.dp
 import androidx.lifecycle.compose.collectAsStateWithLifecycle
-import androidx.lifecycle.viewmodel.compose.viewModel
 import bg.zahov.app.data.model.SetType
 import bg.zahov.app.ui.theme.FitnessTheme
+import bg.zahov.app.ui.workout.add.AddTemplateWorkoutViewModel
 import bg.zahov.fitness.app.R
 import kotlinx.coroutines.delay
 
 @Composable
 fun WorkoutScreen(
-    workoutViewModel: WorkoutViewModel = viewModel(),
+    isWorkoutScreen: Boolean = true,
```

Comment:
Create two separated Screen composables - WorkoutScreen and AddTemplateWorkoutScreen for example. Do not use flags for different  screens. You can extract the common composable in another file and the calls it from screen composable

Reply:
I created 2 different composables and in WorkoutScreenContent to not use a flag I added a content composable that way each screen can decide what it wants to show and we dont need to have optional callbacks